### PR TITLE
long term token regex.

### PR DESCRIPTION
**Pull request by qijieye**
## Change
*(What has changed?)*
regex of card token

## Rationale
*(Why was the change done?)*
long term token can't be able to match current pattern

## Impact
*(What is the impact of this change?)*
encrypted length can be 40 or 48.
short term's encrypted length is 48
long term is 40

## Risk
- [x] The change does not increase the risk to the system and does therefore not require any extra risk analysis.
- [ ] A separate risk analysis has been performed and is linked below.

## Rollback
- [ ] Rollback is performed by reverting the merge and redeploy.
- [ ] Rollback of this change requires special actions as outlined below.

### CODE REVIEW

File: Card/index.spec.ts [hunk 1]
```diff
@@ -120,6 +120,10 @@ describe("@pax2pay/model.Card", () => {
 		expect(model.Card.Token.is(token)).toBeTruthy()
 		token = "4567897890/16/0221/220115/hgUVP-7IwsW963dqwG0sop71sE4F5rCMJ85DJYnwWeGVYdXp/EfyBTD89WFZOfGGq4UGg-w"
 		expect(model.Card.Token.is(token)).toBeTruthy()
```
Comment (simonmika) [inline Card/index.spec.ts:122]: This is a long term token that already worked. Why what is different with the other once?

File: Card/index.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export namespace Card {
 	export type Token = string
 	export namespace Token {
 		export const pattern =
-			/^(\d{10,12})\/(\d{2})\/([0-1]\d{3})\/(([0-2]\d[0-6][02468])|(((([02468][048])|([13579][2,6]))(02((0[1-9])|([1-2]\d))))|(\d{2}((((0[13578])|(1[02]))((0[1-9])|([1-2]\d)|(3[01])))|(02((0[1-9])|(1\d)|(2[0-8])))|(((0[469])|(11))((0[1-9])|([1-2]\d)|(30)))))))\/[a-z,A-Z,0-9\-_]{48}\/[a-z,A-Z,0-9\-_]{22}(\/(pan|csc|expires|month|year|masked))?$/
+			/^(\d{10,12})\/(\d{2})\/([0-1]\d{3})\/(([0-2]\d[0-6][02468])|(((([02468][048])|([13579][2,6]))(02((0[1-9])|([1-2]\d))))|(\d{2}((((0[13578])|(1[02]))((0[1-9])|([1-2]\d)|(3[01])))|(02((0[1-9])|(1\d)|(2[0-8])))|(((0[469])|(11))((0[1-9])|([1-2]\d)|(30)))))))(\/[a-z,A-Z,0-9\-_]{48}|\/([a-z,A-Z,0-9\-_]{40}))\/[a-z,A-Z,0-9\-_]{22}(\/(pan|csc|expires|month|year|masked))?$/
```
Comment (simonmika) [inline Card/index.ts:60]: Use a range instead: `{20,60}` or something. Makes it more forgiving and shortens the expression.

**Review by simonmika — CHANGES_REQUESTED** [review]

File: Card/index.spec.ts (hunk 1 above)
Comment (qijieye) [reply Card/index.spec.ts:122]: encrypted length is different
current short term's encrypted length is 48 & long term's is 40
but in unit test, they are all 48, 
that's why we failed when validate long term token

File: Card/index.ts (hunk 2 above)
Comment (qijieye) [reply Card/index.ts]: i use a range {40,48}

**Review by simonmika — APPROVED** [review]

**Pull request merged**